### PR TITLE
Hooks + Routing

### PR DESCRIPTION
**Pull request by danillouz**
I switched to React hooks, works really nice and is simpler than using traditional (Class) Stateful Components. They are still in alpha, but are recommended to be used if possible:

> Should I use Hooks, classes, or a mix of both?
> When you’re ready, we’d encourage you to start trying Hooks in new components you write. Make sure everyone on your team is on board with using them and familiar with this documentation. We don’t recommend rewriting your existing classes to Hooks unless you planned to rewrite them anyway (e.g. to fix bugs).

You can read more about them if you like [here](https://reactjs.org/docs/hooks-intro.html)

As for routing I used React router. I refactored the Root component into multiple smaller ones, but haven't split them (yet). I'd like to have a better understanding first of the logic, side effects and how a service layer would fit in best. It's a but different with hooks, but shouldn't be a problem. I.e., I'd usually use a HoC or Render Props Component to abstract the API layer (for example to fetch data). But I believe I should write a custom hook/effect for it now.
Hooks also offer a builtin reducer, which is also interesting to look at.

The routing works pretty well if I say so myself. 😅 Try it when you have the time. What's also cool is to play with the viewport (making is larger/smaller), the UI will adapt and show/hide elements, optimized for the screen. 💪

### CODE REVIEW

**Comment by danillouz** [comment]
I found some time to add the data layer and do some refactoring. 😂

File: src/hooks.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  const [error, setError] = useState(null);
+  const [itemsMap, setItems] = useState({});
+
+  const fetch = () => {
+    return new Promise(resolve => {
+      setTimeout(() => {
+        const res = {
+          ok: true,
+          status: 200,
+          statusText: 'KAPUUT',
```
Comment (dinostheo) [inline src/hooks.js]: :)
I guess that's left your testing

File: src/components/Status.jsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  return (
+    <Fragment>
+      {isSmall && (
+        <div id="breadcrumb">
+          <Link to={backRoute}>
+            <button className="back">back</button>
+          </Link>
+
+          <span>{item.name}</span>
```
Comment (dinostheo) [inline src/components/Status.jsx]: Based on the data you have in this PR works well. Using the data that I just merged in master, it won't work. Adjust accordingly. 
Also replace the mock-data.json with the "server response" one, which is a json array like the one you use here.

**Comment by danillouz** [comment]
Cool, now I can pull in your changes. Will do that, modify the components to work with the new data structure and update the PR.

File: src/hooks.js (hunk 1 above)
Comment (danillouz) [reply src/hooks.js]: 😂 you can change the `ok` prop to `false` to trigger the error view, it will use the `statusText` for the error message

**Pull request merged**